library/proc_macro/src/bridge/buffer.rs

```diff
@@ -6,37 +6,37 @@ use std::ops::{Deref, DerefMut};
 use std::slice;
 
 #[repr(C)]
-pub struct Buffer<T: Copy> {
-    data: *mut T,
+pub struct Buffer {
+    data: *mut u8,
```

Comment:
For the record, I think the reason I made this generic was to avoid accidentally doing the wrong thing because the types *happened* to match up (especially since there's `unsafe` code here) but I'll review this carefully and hopefully this is fine.

Reply:
Would any type other than `u8` make sense here? Doesn't seem like it, especially given that `Buffer<u8>` was hardcoded in a bunch of places anyway.

Reply:
That's not what I meant, but rather that writing parametric code can sometimes help avoid making mistakes (which can be catastrophic in `unsafe` code).

It doesn't make sense at all from the perspective of whether e.g. a C++ container might be templated or not, but in does in Rust because of the type-checking in the generic form.

Anyway it doesn't matter much, I was just explaining that `Foo<T>` can still make sense even if it's only ever used as a single `Foo<Concrete>` and the generic nature "isn't taken advantage of".

Hopefully we can remove this `Buffer` abstraction and move more to e.g. a model closer to read/write syscalls (or `io::{Read,Write}` traits I suppose, at a higher level).